maker/src/health.rs

```diff
@@ -0,0 +1,120 @@
+// TODO: Add health collection metrics
+// This can be later used to add health metrics to the health endpoint
+
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use reqwest::Client;
+use reqwest::StatusCode;
+use reqwest::Url;
+use std::time::Duration;
+use tokio::sync::watch;
+
+/// Health status of a service
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum ServiceStatus {
+    #[default]
+    Unknown,
+    Online,
+    Offline,
+}
+
+/// Health monitoring for the node
+///
+/// Simple endpoint querying is handled by provided configuration, for more complex health checks
+/// the transmitters are exposed to be plugged in the services that need to publish their health
+/// status.
+
+#[derive(Clone)]
+pub struct Health {
+    /// Coordinator HTTP API status
+    coordinator_rx: watch::Receiver<ServiceStatus>,
+    /// Orderbook websocket stream status
+    orderbook_rx: watch::Receiver<ServiceStatus>,
+}
+
+/// Transmitters that need to be plugged in the services that need to publish their health status.
+pub struct Tx {
+    pub orderbook: watch::Sender<ServiceStatus>,
+    pub coordinator: watch::Sender<ServiceStatus>,
+}
+
+impl Health {
+    pub fn new() -> (Self, Tx) {
+        let (orderbook_tx, orderbook_rx) = watch::channel(ServiceStatus::Unknown);
+        let (coordinator_tx, coordinator_rx) = watch::channel(ServiceStatus::Unknown);
+
+        (
+            Self {
+                coordinator_rx,
+                orderbook_rx,
+            },
+            Tx {
+                orderbook: orderbook_tx,
+                coordinator: coordinator_tx,
+            },
+        )
+    }
+
+    // TODO: Any ideas of the most useful way of sending this information?
+    // - perhaps a serialized struct (json) with the status of each service?
+    pub fn get_health(&self) -> Result<String> {
+        let mut health_info = String::new();
+        health_info.push_str(&format!(
+            "Coordinator: {:?}\n",
+            self.get_coordinator_status()
+        ));
+        health_info.push_str(&format!("Orderbook: {:?}\n", self.get_orderbook_status()));
+
+        if self.get_coordinator_status() == ServiceStatus::Online
+        // TODO: Uncomment this line when we retrieve the orderbook status
+        // && (self.get_orderbook_status() == ServiceStatus::Online)
+        {
+            Ok(health_info)
+        } else {
+            bail!("Status: ERROR\n + {health_info}");
+        }
+    }
+
+    pub fn get_coordinator_status(&self) -> ServiceStatus {
+        *self.coordinator_rx.borrow()
+    }
+
+    pub fn get_orderbook_status(&self) -> ServiceStatus {
+        *self.orderbook_rx.borrow()
+    }
+    // TODO: Add bitmex status? would anything else be useful here?
```

Comment:
Should this rather be captured in a ticket?

Reply:
this was not in the original scope of the ticket, so I have left it as a comment first - wasn't sure whether we wanted it! 
it seems it can be addressed rather quickly, so I'll add support for it in this PR.